routes/host_routes.py

```diff
@@ -3,18 +3,39 @@
 from lib.user_repository import UserRepository
 from lib.space_repository import SpaceRepository
 from lib.user import User
+from lib.space import Space
 import os
 import sys
 
+
 def get_host_routes(app):
 
-    @app.route('/host-listings', methods=['GET'])
-    
+    @app.route("/host-listings", methods=["GET"])
     def get_host_listings_page():
         # shows all listings for signed in host
-        user_id = session['user_id']
+        user_id = session["user_id"]
         connection = get_flask_database_connection(app)
         repository = SpaceRepository(connection)
         spaces = repository.get_by_user_id(user_id)
 
-        return render_template('host_listings.html', spaces=spaces)
\ No newline at end of file
+        return render_template("host_listings.html", spaces=spaces)
+
+    @app.route("/create-space", methods=["GET"])
+    def get_new_space_page():
+        return render_template("create_space.html")
+
+    @app.route("/create-space", methods=["POST"])
+    def create_new_space():
+        connection = get_flask_database_connection(app)
+        repository = SpaceRepository(connection)
+        space = Space(
+            None,
+            request.form["name"],
+            request.form["description"],
+            request.form["price_per_night"],
+            request.form["availability_start"],
```

Comment:
Should we add some logic to ensure dates are passed in in the correct format here?

Reply:
We can set the input type in html to date so it you have to choose a date from a calender, ive updated it and will push the changes now

Reply:
Perfect. When that's there I'll approve =]